samtranslator/swagger/swagger.py

```diff
@@ -781,6 +783,46 @@ def _set_method_apikey_handling(self, path, method_name, apikey_required):
             if security != existing_security:
                 method_definition["security"] = security
 
+    def add_request_validator_to_method(self, path, method_name, validate_body=False, validate_parameters=False):
+        """
+        Adds request model body parameter for this path/method.
+
+        :param string path: Path name
+        :param string method_name: Method name
+        :param bool validate_body: Add validator parameter on the body
+        :param bool validate_parameters: Validate request
+        """
+
+        normalized_method_name = self._normalize_method_name(method_name)
+        validator_name = SwaggerEditor.get_validator_name(validate_body, validate_parameters)
+
+        # Creating validator
+        request_validator_definition = {
+            validator_name: {"validateRequestBody": validate_body, "validateRequestParameters": validate_parameters}
+        }
+        if not self._doc.get(self._X_APIGW_REQUEST_VALIDATORS):
+            self._doc[self._X_APIGW_REQUEST_VALIDATORS] = {}
+
+        if not self._doc[self._X_APIGW_REQUEST_VALIDATORS].get(validator_name):
+            # Adding only if the validator hasn't been defined already
+            self._doc[self._X_APIGW_REQUEST_VALIDATORS].update(request_validator_definition)
+
+        # It is possible that the method could have two definitions in a Fn::If block.
+        # for method_definition in self.get_method_contents(self.get_path(path)[normalized_method_name]):
+        for method_name, method in self.get_path(path).items():
+
+            # Adding it to only given method to the path
+            if method_name == normalized_method_name:
```

Comment:
We may have an issue if the `method_name` is not normalized (in the template).

Also, this name is conflicting with the function's parameter `method_name` so I think we should find another name.

Reply:
ok, just pushed what I think make more sense now. Normalizing the method_name that comes from the template before check if the method definition is the one we want to add the validator.